### PR TITLE
kernel function information for hip

### PR DESCRIPTION
**Pull request by CRobeck**

### CODE REVIEW

File: include/RAJA/policy/hip/kernel/internal.hpp [hunk 1]
```diff
@@ -410,6 +410,26 @@ using hip_statement_list_executor_t = HipStatementListExecutor<
...
+  static unsigned int getKernelRegisters(
+  	hipFuncAttributes const &hip_func_attributes) {
+	 return hip_func_attributes.numRegs;
+   }
+
+
+static hipFuncAttributes get_hip_func_attributes(void const *kernel_func) {
+    static hipFuncAttributes attr = [=]() {
+      hipFuncAttributes attr;
+      hipFuncGetAttributes(&attr, kernel_func);
```
Comment (MrBurmark) [inline include/RAJA/policy/hip/kernel/internal.hpp:427]: Should this be error checked?

File: include/RAJA/policy/hip/kernel/internal.hpp [hunk 2]
```diff
@@ -410,6 +410,26 @@ using hip_statement_list_executor_t = HipStatementListExecutor<
...
 
+struct HIPKernelLaunchFuncData {
+  static unsigned int getKernelScratchBytes(
+      hipFuncAttributes const &hip_func_attributes) {
+    return hip_func_attributes.localSizeBytes;
+  }
+  static unsigned int getKernelRegisters(
+  	hipFuncAttributes const &hip_func_attributes) {
+	 return hip_func_attributes.numRegs;
+   }
```
Comment (MrBurmark) [inline include/RAJA/policy/hip/kernel/internal.hpp:421]: spacing

File: include/RAJA/policy/hip/kernel/internal.hpp (hunk 1 above)
Comment (CRobeck) [reply include/RAJA/policy/hip/kernel/internal.hpp:427]: I was looking for a HIP_SAFE_CALL macro defined somewhere. Do we have one?

Comment (MrBurmark) [reply include/RAJA/policy/hip/kernel/internal.hpp:427]: Take a look here https://github.com/LLNL/RAJA/blob/develop/include/RAJA/policy/hip/raja_hiperrchk.hpp#L45

**Comment by artv3** [comment]
Cool! Is this something that could also be used within the RAJA Teams framework?

**Pull request closed without merging**